src/main/java/teammates/ui/controller/StudentCommentsPageAction.java

```diff
@@ -153,7 +158,49 @@ private void setNextPageLink(List<CourseAttributes> courses, int currentIndex){
                 }
             }
         }
-        return feedbackResultBundles;
+        return sortFeedbackBundlesByLastCommentDate(feedbackResultBundles);
+    }
+    
+    /**
+     * @param feedbackResultBundles     The map of FeedbackSessionResultsBundle which is to be sorted 
+     * @return                          Returns a sorted map(LinkedHashMap) of FeedbackSessionResultsBundle,
+     *                                  where the sessions with the latest comment come first.
+     */
+    private Map<String, FeedbackSessionResultsBundle> sortFeedbackBundlesByLastCommentDate(
+            Map<String, FeedbackSessionResultsBundle> feedbackResultBundles) {
+        
+        //this maps all keys of the original map to their corresponding last comment date
+        //this is used during sorting
+        final HashMap<String, Date> lastCommentDateMap = new HashMap<String, Date>();
+        for (Entry<String, FeedbackSessionResultsBundle> entry : feedbackResultBundles
+                .entrySet()) {
+            lastCommentDateMap.put(entry.getKey(), entry.getValue()
+                    .getLastCommentDate());
+        }
+        
+        List<String> list = new ArrayList<String>(
+                feedbackResultBundles.keySet());
+        Collections.sort(list, new Comparator<String>() {
+
+            @Override
+            public int compare(String o1, String o2) {
+                Date date1 = lastCommentDateMap.get(o1);
+                Date date2 = lastCommentDateMap.get(o2);
+                if(date1 == null) {
+                    return 1;
+                }
+                if(date2 == null) {
+                    return -1;
+                }
+                return date2.compareTo(date1);
```

Comment:
Yeah they are ordered in descending order of session's latest comment date. So the session with the latest comment comes first. I interpreted
"Comments for sessions, ordered in reverse chronological order of the sessions"
to mean this only in the context of the issue #2679. Maybe @damithc can clarify.